instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/internal/BedrockRuntimeImpl.java

```diff
@@ -813,11 +828,20 @@ private static void recordInvokeModelRequestEvents(
         }
         List<ContentBlock> parsedContentBlocks = new ArrayList<>();
         for (Document contentBlock : content.asList()) {
-          Document text = contentBlock.asMap().get("text");
-          if (text == null || !text.isString()) {
-            continue;
+          ContentBlock parsed;
+          switch (modelFamily) {
+            case AMAZON_NOVA:
+              parsed = parseAmazonNovaContentBlock(contentBlock);
+              break;
+            case ANTHROPIC_CLAUDE:
+              parsed = parseAnthropicClaudeContentBlock(contentBlock);
+              break;
+            default:
+              parsed = null;
+          }
+          if (parsed != null) {
+            parsedContentBlocks.add(parsed);
```

Comment:
This block appears twice, could consider extracting to a method. Up to you to decide whether that makes sense.